ui/admin/app/routes/scopes/scope/sessions/index.js

```diff
@@ -4,5 +4,171 @@
  */
 
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
 
-export default class ScopesScopeSessionsIndexRoute extends Route {}
+export default class ScopesScopeSessionsIndexRoute extends Route {
+  // =services
+
+  @service store;
+
+  // =attributes
+
+  queryParams = {
+    search: {
+      refreshModel: true,
+      replace: true,
+    },
+    users: {
+      refreshModel: true,
+      replace: true,
+    },
+    targets: {
+      refreshModel: true,
+      replace: true,
+    },
+    status: {
+      refreshModel: true,
+      replace: true,
+    },
+    page: {
+      refreshModel: true,
+    },
+    pageSize: {
+      refreshModel: true,
+    },
+  };
+
+  allUsers;
+  allSessions;
+  allTargets;
+
+  /**
+   * Loads all sessions under the current scope and encapsulates them into
+   * an array of objects containing their associated users and targets.
+   * @return {Promise{[{sessions: [SessionModel], allSessions: [SessionModel], allUsers: [UserModel], allTargets: [TargetModel], totalItems: number}]}}
+   */
+  async model({ search, users, targets, status, page, pageSize }) {
+    const { id: scope_id } = this.modelFor('scopes.scope');
+    const filters = {
+      status: [],
+      user_id: [],
+      target_id: [],
+    };
+    users.forEach((user) => {
+      filters.user_id.push({ equals: user });
+    });
+    targets.forEach((target) => {
+      filters.target_id.push({ equals: target });
+    });
+    status.forEach((item) => {
+      filters.status.push({ equals: item });
+    });
+
+    const queryOptions = {
+      scope_id,
+      recursive: true,
+      query: { search, filters },
+      page,
+      pageSize,
+    };
+
+    const sessions = await this.store.query('session', queryOptions);
+    const totalItems = sessions.meta?.totalItems;
+
+    // Query all sessions, users, and targets for defining filtering values if entering route for the first time
+    if (!this.allSessions) {
+      await this.getAllSessions(scope_id);
+    }
+    if (!this.allUsers) {
+      await this.getAllUsers();
+    }
+    if (!this.allTargets) {
+      await this.getAllTargets(scope_id);
+    }
+
+    return {
+      sessions,
+      allSessions: this.allSessions,
+      allUsers: this.allUsers,
+      allTargets: this.allTargets,
+      totalItems,
+    };
+  }
+
+  /**
+   * Get all the sessions but only load them once when entering the route.
+   * @param scope_id
+   * @returns {Promise<void>}
+   */
+  async getAllSessions(scope_id) {
+    const allSessionsQuery = {
+      scope_id,
+      recursive: true,
+    };
+    this.allSessions = await this.store.query('session', allSessionsQuery, {
+      pushToStore: false,
+    });
+  }
+
+  /**
+   * Get all the users but only load them once when entering the route.
+   * @returns {Promise<void>}
+   */
+  async getAllUsers() {
```

Comment:
I would make these names a bit more descriptive since I do see that they're only pulling in users and targets that are on a session, I thought these were originally pulling in _all_ users and targets based on the function names

Reply:
Great point. I will update now.

Reply:
Updated.